WordPress/src/main/java/org/wordpress/android/ui/uploads/PostUploadNotifier.java

```diff
@@ -33,62 +36,161 @@
+            // check if special case for ONE media item
+            if (sNotificationData.totalMediaItems == 1) {
+                mNotificationBuilder.setContentTitle(buildNotificationTitleForMedia());
+                mNotificationBuilder.setContentText(buildNotificationSubtitleForMedia());
+            } else {
+                mNotificationBuilder.setContentTitle(buildNotificationTitleForMixedContent());
+                mNotificationBuilder.setContentText(buildNotificationSubtitleForMedia());
+            }
+
+        } else if (sNotificationData.totalMediaItems == 0 && sNotificationData.totalPostItems > 0) {
+            // only Post / Pages are being uploaded
+            // check if special case for ONE Post
+            if (sNotificationData.totalPostItems == 1) {
+                mNotificationBuilder.setContentTitle(buildNotificationTitleForPost(post));
+                mNotificationBuilder.setContentText(buildNotificationSubtitleForPost(post));
+            } else {
+                mNotificationBuilder.setContentTitle(buildNotificationTitleForMixedContent());
+                mNotificationBuilder.setContentText(buildNotificationSubtitleForPosts());
+            }
+        } else {
+            // mixed content (Post/Pages and media) is being uploaded
+            mNotificationBuilder.setContentTitle(buildNotificationTitleForMixedContent());
+            mNotificationBuilder.setContentText(buildNotificationSubtitleForMixedContent());
         }
-        int notificationId = (new Random()).nextInt() + post.getLocalSiteId();
+    }
 
-        NotificationData notificationData = new NotificationData();
-        notificationData.notificationId = notificationId;
-        sPostIdToNotificationData.put(post.getId(), notificationData);
-        mService.startForeground(notificationId, mNotificationBuilder.build());
+    private synchronized void startOrUpdateForegroundNotification(@Nullable PostModel post) {
+        updateNotificationBuilder(post);
+        if (sNotificationData.notificationId == 0) {
+            sNotificationData.notificationId = (new Random()).nextInt();
+            mService.startForeground(sNotificationData.notificationId, mNotificationBuilder.build());
+        } else {
+            // service was already started, let's just modify the notification
+            doNotify(sNotificationData.notificationId, mNotificationBuilder.build());
+        }
+    }
+
+    // Post could have initial media, or not (nulable)
+    void addPostInfoToForegroundNotification(@NonNull PostModel post, @Nullable List<MediaModel> media) {
+        sNotificationData.totalPostItems++;
+        if (post.isPage()) {
+            sNotificationData.totalPageItemsIncludedInPostCount++;
+        }
+        if (media != null) {
+            addMediaInfoToForegroundNotification(media);
+        }
+        startOrUpdateForegroundNotification(post);
+    }
+
+    void addMediaInfoToForegroundNotification(@NonNull List<MediaModel> media) {
```

Comment:
I'm seeing a number of warnings in these routines about unused parameters. Is this for future use?

<img width="642" alt="screen shot 2017-10-17 at 5 15 00 pm" src="https://user-images.githubusercontent.com/3903757/31689918-126dd9dc-b35f-11e7-849f-bfdc5b82a992.png">

Reply:
Sorry I forgot to mention this in the description - yeah it is intended so I remember for the PRs to come (that will be merged against this branch), that will actually either make use of these params or finally remove them 👍